app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt

```diff
@@ -39,17 +43,37 @@ class STTFragment : Fragment() {
         val rootView = inflater.inflate(R.layout.fragment_sttframe, container, false)
         if (thisActivity is ChatActivity)
             thisActivity.fabsetting.hide()
-        promptSpeechInput()
         setupCommands(rootView)
         return rootView
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        Handler().post {
+            textToSpeech = TextToSpeech(requireContext(), TextToSpeech.OnInitListener { status ->
+                if (status != TextToSpeech.ERROR) {
+                    val locale = textToSpeech?.language
+                    textToSpeech?.language = locale
+                    if (!PrefManager.getBoolean(R.string.used_voice, false)) {
+                        textToSpeech?.speak(getString(R.string.voice_welcome), TextToSpeech.QUEUE_FLUSH, null)
+                        PrefManager.putBoolean(R.string.used_voice, true)
+                        Handler().postDelayed({
+                            promptSpeechInput()
+                        }, 1500) // Starting speech engine after welcome message is spoken
+                    } else {
+                        promptSpeechInput()
+                    }
+                }
+            })
+        }
```

Comment:
All this logic should be placed in the presenter. The view should only know what to show to the user.

Reply:
There is no presenter class made for STT Fragment. Also, I found that the logic of the other functions was also written in the fragment. Hence, I wrote the code in the fragment itself.

Reply:
@arundhati24 what should I do? Shall I make presenter class for the fragment or just keep it here>

Reply:
Clear all references to Handler onDestroyView